lib/addons/karpenter/index.ts

```diff
@@ -519,10 +519,14 @@ export class KarpenterAddOn extends HelmAddOn {
         }
 
         if (semver.gte(version, "0.32.0") && installCRDs){
+            // Since v0.35.0, Karpenter OCI tags and Helm chart version are now valid semantic versions,
+            // meaning that the v prefix from the git tag has been removed and they now follow the x.y.z pattern
+            // But the CRDs URL still uses the git tag format, so we need to check for the version and adjust the URL accordingly
+            let _version = (semver.gte(version, "0.35.0")) ? `v${version}` : version;
```

Comment:
@vumdao alsmost there. 
I looked through the branches of Karpenter and it does not look like the version convention with `v` is something specifically enforced before 0.35.0. 

Let's add an explicit check: `if(!version.startsWith('v') then add `v` into the version in URL, to avoid potential mismatches.